src/plugins/action_update_review_assignments/src/updateReviewAssignmentsStatus.ts

```diff
@@ -0,0 +1,52 @@
+import { AssignmentStatus } from './types'
+import databaseMethods from './databaseMethods'
+
+module.exports['updateReviewAssignmentsStatus'] = async function (parameters: any, DBConnect: any) {
+  const db = databaseMethods(DBConnect)
+  console.log('Updating review assignment statuses...')
+  try {
+    const { reviewAssignmentId, trigger } = parameters
+    // NB: reviewAssignmentId comes from record_id on TriggerPayload when
+    // triggered from review_assignment table
+    const {
+      application_id: applicationId,
+      stage_number: stageNumber,
+      level: reviewLevel,
+    } = await db.getReviewAssignmentById(reviewAssignmentId)
+
+    const otherReviewAssignments = await db.getMatchingReviewAssignments(
+      reviewAssignmentId,
+      applicationId,
+      stageNumber,
+      reviewLevel
+    )
+
+    const reviewAssignmentUpdates = await Promise.all(
+      otherReviewAssignments.map(async (reviewAssignment: any) => {
+        const { id, status } = reviewAssignment
+        return {
+          id,
+          status: trigger === 'onReviewSelfAssign' ? AssignmentStatus.SELF_ASSIGNED_OTHER : status,
```

Comment:
Is this the first time we are checking 'trigger' inside an action ? Does it mean this action is specific to that trigger ?

Reply:
> Is this the first time we are checking 'trigger' inside an action ?

Yes, I think it is, which is why I was hoping to avoid it, but there doesn't seem to be any other way to distinguish between an assignment and a self-assignment.


> Does it mean this action is specific to that trigger ?

Technically, no -- but in practice yes, since it wouldn't make sense to use it in other contexts.